LawsonMcCoy/Fading_Light_Narcissus_Lux
# Dialogue System

Instructions on how to test the Dialogue System:

1. **Open Dialogue Test Scene.** This scene should contain a DialogueUI object (that is currently _inactive_), a TestCapsule object, and a DialogueManager system. The TestCapsule should be a green capsule in the middle of a brown plane.
2. **Create a Character Scriptable Object**. Go to Assets -> ScriptableObjects -> Characters and right-click -> Create -> Character to create one. **IMPORTANT**: Make sure to set the Full Name property to its name in the inspector.
3. **Create a Dialogue Scriptable Object**. Go to Assets -> ScriptableObjects -> Dialogues and right-click -> Create -> Dialogue to create one. On the inspector, add a couple of sentences to the dialogue by pressing the "+" button. Every sentence should have a speaker and a text field to put the sentences in. Make sure you assign a speaker (by using the Character you made earlier) and the sentences that which the speaker will say.
4. In the DialogueManager, grab the Dialogue Scriptable Object you made and drag it to the appropriate field named Dialogue. Note that the DialogueManager should have references to the dialogueUI, the speakerName text field, the textField for the sentence, the dialogue (in which you already dragged and set), and a typeSpeed that controls the speed in which the letters of the sentences are being typed.
5. **Play the game**. Click the TestCapsule object in the middle of the scene and it should trigger the dialogue you have set. **IMPORTANT**: Currently, press **SPACE** to advance the dialogue.
6. Once the dialogue has finished, the UI should dissappear.
